frontend/taipy-gui/base/src/cookieHandler.ts

```diff
@@ -51,12 +45,9 @@ export class CookieHandler {
     }
 
     addBeforeUnloadListener() {
-        window.deleteCookie = () => {
-            axios.delete("taipy-rh");
-        };
         window.addEventListener("beforeunload", () => {
             localStorage.removeItem(TAIPY_RESOURCE_HANDLER);
-            window.deleteCookie();
+            this.deleteCookie();
```

Comment:
are you sure this is valid in this context ?
I would have saved it in the scope with `const self = this;`

Reply:
I think arrow function (method) are automatically bound ?

Reply:
It is still working per my testing. I will try your method later on if it breaks somehow and I like your proposal very much too!